lib/rubygems/platform.rb

```diff
@@ -75,6 +76,7 @@ def initialize(arch)
 
       @cpu = case cpu
              when /i\d86/ then 'x86'
+             when /universal\.(.*)/ then $1
```

Comment:
I think `universal` included in `Gem::Platform.local` is incorrect, so stripping `universal.` leaving only `x86_64` or `arm64e` as the CPU value is correct as implemented here.

Perhaps this regexp should also be anchored, `/\A universal\.(.*)/`?